Python/bytecodes.c

```diff
@@ -411,28 +411,28 @@ dummy_func(
             // BINARY_OP_INPLACE_ADD_UNICODE,  // See comments at that opcode.
         };
 
-        op(_GUARD_BOTH_INT, (left, right -- left, right)) {
+        guard op(_GUARD_BOTH_INT, (left, right -- left:  ~(PYINT_TYPE), right:  ~(PYINT_TYPE))) {
```

Comment:
Why the `~` prefix to the type. That implies that `left` is not a "PYINT_TYPE" which is the opposite of what happens.
Can you use `int` or `PyLong_Type`, rather than introducing new names?

Reply:
Perhaps use an `&` prefix and the C names. That way `left:  ~(PYINT_TYPE)` becomes `left: &PyLong_Type` which matches the C syntax for the pointer to the `int` type.

Reply:
I'll use the & prefix but not the existing names. The main reason is that we have types that aren't represented by either the C or Python types. An example is `GUARD_DORV_VALUES_TYPE`. That's a shape check that makes sure it's a values layout. However it does not correspond to another `Py_Type` or C type.